cirq-core/cirq/devices/fidelity.py

```diff
@@ -0,0 +1,216 @@
+
+        self._t1 = t1
+        self._p = decay_constant
+        self._p00 = p00
+        self._p11 = p11
+        self._pauli_error = pauli_error
+        self._xeb = xeb_fidelity
+
+        if self._pauli_error is not None:
+            self._p = self.pauli_error_to_decay_constant()
+            self._xeb = self.pauli_error_to_xeb_fidelity()
+        elif self._p is not None:
+            self._pauli_error = self.decay_constant_to_pauli_error()
+            self._xeb = self.decay_constant_to_xeb_fidelity()
+        elif self._xeb is not None:
+            self._p = self.xeb_fidelity_to_decay_constant()
+            self._pauli_error = self.decay_constant_to_pauli_error()
+
+    @property
+    def decay_constant(self):
+        return self._p
+
+    @property
+    def p00(self):
+        return self._p00
+
+    @property
+    def p11(self):
+        return self._p11
+
+    @property
+    def pauli_error(self):
+        return self._pauli_error
+
+    @property
+    def t1(self):
+        return self._t1
+
+    @property
+    def xeb(self):
+        return self._xeb
+
+    def decay_constant_to_xeb_fidelity(self, N: int = 4):
+        if self._p is not None:
+            return 1 - ((1 - self._p) * (1 - 1 / N))
+        return None
+
+    def decay_constant_to_pauli_error(self, N: int = 2):
+        if self._p is not None:
+            return (1 - self._p) * (1 - 1 / N / N)
+        return None
+
+    def pauli_error_to_xeb_fidelity(self, N: int = 4):
+        if self._pauli_error is not None:
+            self._p = 1 - (self._pauli_error / (1 - 1 / N))
+            return self.decay_constant_to_xeb_fidelity()
+        return None
+
+    def pauli_error_to_decay_constant(self, N: int = 2):
+        if self._pauli_error is not None:
+            return 1 - (self._pauli_error / (1 - 1 / N / N))
+        return None
+
+    def xeb_fidelity_to_decay_constant(self, N: int = 4):
+        if self._xeb is not None:
+            return 1 - (1 - self._xeb) / (1 - 1 / N)
+        return None
+
+    def pauli_error_from_t1(self, t: float, t1: float):
+        t2 = 2 * t1
+        return (1 - np.exp(-t / t2)) / 2 + (1 - np.exp(-t / t1)) / 4
+
+    def pauli_error_from_depolarization(self, t: float):
+        if self._t1 is not None:
+            return self._pauli_error - self.pauli_error_from_t1(t, self._t1)
+        else:
+            return self._pauli_error
+
+    def rb_pauli_error(self, N: int = 2):
+        if self._p is not None:
+            return (1 - self._p) * (1 - 1 / N ** 2)
+        return None
+
+    def rb_average_error(self, N: int = 2):
+        if self._p is not None:
+            return (1 - self._p) * (1 - 1 / N)
+        return None
+
+
+class NoiseModelFromFidelity(cirq.NoiseModel):
+    def __init__(self, fidelity: Fidelity) -> None:
+        if fidelity is not None:
+            self._fidelity = fidelity
+        else:
+            raise ValueError('A fidelity object must be specified')
+
+    def get_duration(self, gate):
```

Comment:
Please add a TODO here to pull the gate durations from cirq_google, or to allow users to pass them in.

Reply:
Also, for clarity this should be `get_duration_ns`.